pepper-apis/src/main/java/org/broadinstitute/ddp/route/AddProfileRoute.java

```diff
@@ -56,7 +57,7 @@ public Object handle(Request request, Response response, Profile profile) throws
                         new ApiError(ErrorCodes.INVALID_LANGUAGE_PREFERENCE, "Invalid preferred language"));
             }
 
-            UserProfileDao profileDao = handle.attach(UserProfileDao.class);
+            UserProfileDao profileDao = new UserProfileCachedDao(handle);
             boolean exists = profileDao.findProfileByUserGuid(userGuid).isPresent();
```

Comment:
Wherever we modify data that might affect cache coherency, we use our wrapped DAOs.